### DISCUSSION TITLE
Allow for multiple languages websites #60

### DISCUSSION THREAD

**Discussion by hubyhuby** [GetPublii/Publii#1586] [General]
Any roadmap on the multiple languages websites support ?
In 0.46 ? 0.47 ?

Even if it is beta , I am willing to test it;
I would make a dummy website and publish it online.

https://github.com/GetPublii/Publii/issues/60

I have made several websites, and love Publii. Now one of my website really needs English + French ;)

**Comment by szektor78** [comment]
I would love to see that as well.

**Comment by gppoletti2** [comment]
Hello! I'm using publii from short time.
I will need the multilanguage feature.
I'm an old c++ programmer but I have very little experience in website html css and the like.
But I have an idea on the multilanguage problem. See if it sound good for you.

The website created with publii are static by definition. It means that every page exixts inside the website folder.
Let's say that I want two languages english (en) and polish (pl).
I can create with publii two websites (website_en and website_pl) with exactly the same structure, same pages, same posts, one is the mirror of the other except for the text inside that is translated.

The remote path are /myweb/en and /myweb/pl, and the page structure below is exactly the same.

Let's say that publii will create (in the upper right corner of every page) a widget (or small menu or whatever) that when clicked triggers a java script procedure. That small procedure simply takes the current URL and replace the "en" text with "pl" (and viceversa) and reload the URL.
I think that it should work with no big changes in the way publii works, just adding the widget that trigger some procedure written in java.
What do you think?

**Reply by szektor78** [reply]
Yes, this works, however imagine if you have 5-6 languages and you have to copy a new post/page to all places each time you add new content. Even with 3 this quickly becomes uncomfortable. (What if you overwrite one? You need a backup. Oh, sh*t I forgot to backup... ;) )

**Reply by candidexmedia** [reply]
@szektor78 What kind of solution are you looking for? What workflow would be best for you?

**Reply by gppoletti2** [reply]
Of course, if you have 10 languages you have to write every post or every page 10 times.
